ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/infraestructure/antlr/visitors/classes/JavaClassVisitor.java

```diff
@@ -0,0 +1,59 @@
+package edu.usb.argos.ASTProcessor.visitor.infraestructure.antlr.visitors.classes;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaParserBaseVisitor;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.classes.ClassIdentity;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.classes.ClassInformation;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.classes.ClassMembers;
+import edu.usb.argos.ASTProcessor.visitor.core.entities.classes.ClassStructure;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassAnalyzerVisitor;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassIdentityCollector;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassMemberCollector;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassStructureCollector;
+import org.antlr.v4.runtime.ParserRuleContext;
+
+public class JavaClassVisitor extends JavaParserBaseVisitor<ClassInformation>
+        implements IClassAnalyzerVisitor<ParserRuleContext> {
+
+    private final IClassIdentityCollector<ParserRuleContext> identityCollector;
+    private final IClassStructureCollector<ParserRuleContext> structureCollector;
+    private final IClassMemberCollector<ParserRuleContext> memberCollector;
+
+    public JavaClassVisitor(
+            IClassIdentityCollector<ParserRuleContext> identityCollector,
+            IClassStructureCollector<ParserRuleContext> structureCollector,
+            IClassMemberCollector<ParserRuleContext> memberCollector) {
+        this.identityCollector = identityCollector;
+        this.structureCollector = structureCollector;
+        this.memberCollector = memberCollector;
+    }
+
+    @Override
+    public ClassInformation visitClass(ParserRuleContext ctx) {
```

Comment:
This method should be split into smaller, builder-specific methods.

Reply:
done